dvc/remote/local/__init__.py

```diff
@@ -222,7 +223,7 @@ def isdir(self, path_info):
         return os.path.isdir(fspath_py35(path_info))
 
     def walk(self, path_info):
-        return os.walk(fspath_py35(path_info))
+        return dvc_walk(fspath_py35(path_info), dvcignore=self.repo.dvcignore)
```

Comment:
The agreed (but not yet implemented) approach to [file utils](https://github.com/iterative/dvc/issues/2137) is to make them Path-like objects aware. So I suggest calling `fspath*()` inside `dvc_walk()`.